pkg/resourcemanager/cosmosdb/sqldatabase/resource_differ.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package sqldatabase
+
+import "github.com/Azure/azure-sdk-for-go/services/cosmos-db/mgmt/2021-03-15/documentdb"
+
+func doResourcesMatch(expected *documentdb.SQLDatabaseResource, actual *documentdb.SQLDatabaseGetPropertiesResource) bool {
+	var expectedIDCoalesced *string
+	if expected != nil {
+		expectedIDCoalesced = expected.ID
+	}
+
+	var actualIDCoalesced *string
+	if actual != nil {
+		actualIDCoalesced = actual.ID
+	}
```

Comment:
I'm not sure this buys you anything - why not just use `expected.ID` where you're using `*expectedIDCoalesced`? Oh crap, is it because ID is a `*string`? Gross. Ok, as you were.